staging/src/k8s.io/api/core/v1/types.go

```diff
@@ -3973,6 +3973,8 @@ type ServiceSpec struct {
 	// The primary use case for setting this field is to use a StatefulSet's Headless Service
 	// to propagate SRV records for its Pods without respect to their readiness for purpose
 	// of peer discovery.
+	// If this is set to true on a Service with a selector, all endpoints derived from the
```

Comment:
I think this whole comment could be redone.  Something like:

"""
publishNotReadyAddresses indicates that any agent which deals with endpoints for a Service should disregard any indications of ready/not-ready.  The primary use case for setting this field is to use a StatefulSet's Headless Service to propagate SRV DNS records for its Pods for the purpose of peer discovery.

The Kubernetes controllers that generate Endpoints and EndpointSlice resources for Services interpret this to mean that all endpoints are considered "ready" even if the Pods themselves are not. Agents which only consume endpoints through the Endpoints or EndpointSlice resources can safely assume this behavior.
"""

Reply:
I think that's better. I took your suggestion with a minor edit to the last sentence to specify that this only applies to K8s generated endpoints, since custom controllers/manually created endpoints may still use ready/not-ready.